pennylane/pauli/utils.py

```diff
@@ -680,24 +672,21 @@ def are_pauli_words_qwc(lst_pauli_words):
     Returns:
         (bool): True if they are all qubit-wise commuting, false otherwise.
     """
-    latest_op_name_per_wire = {}
+    basis = {}
 
-    for op in lst_pauli_words:  # iterate over the list of observables
-        op_names = [op.name] if not isinstance(op.name, list) else op.name
-        op_wires = op.wires.tolist()
+    for op in lst_pauli_words:
+        # just need the first term since it should already be only one pauli word
+        if is_pauli_word(op):
+            pw = next(iter(qml.pauli.pauli_sentence(op)))
+        else:
+            pw = _non_pauli_word_to_dict_form(op)
```

Comment:
2 comments: 
- Firstly, I think it would be best if we move this logic out to a wrapper function to handle operators that aren't pauli words. 
- Secondly, for observables like `qml.Hermitian()` which define a specific matrix, this check is not exhaustive and will cause false negatives (eg. if the hermitian observable on wire 0 is a PauliX matrix, this check will fail if you also apply a PauliX operator to wire 0 because they have different names.).